lib/ide/src/view.rs

```diff
@@ -13,3 +13,45 @@ pub mod temporary_panel;
 pub mod project;
 pub mod layout;
 pub mod text_editor;
+
+
+
+// ========================
+// === KeyboardListener ===
+// ========================
+// This code is temporary and will be replace with FRP keyboard events once it's fully functional.
+
+use wasm_bindgen::prelude::*;
+use basegl::prelude::*;
+
+use basegl::system::web::document;
+use js_sys::Function;
+use web_sys::HtmlElement;
+use web_sys::KeyboardEvent;
+use wasm_bindgen::JsCast;
+
+
+type KeyboardClosure = Closure<dyn FnMut(KeyboardEvent)>;
+
+#[derive(Debug)]
+struct KeyboardListener {
+    callback   : KeyboardClosure,
+    element    : HtmlElement,
+    event_type : String
+}
+
+impl KeyboardListener {
+    fn new(event_type:String, callback:KeyboardClosure) -> Self {
```

Comment:
I don't like how this API requires passing some magic strings.
If their set is closed, I'd consider wrapping wrapped as some kind of enum.
Otherwise,  to avoid `.into` on arguments in call place, consider taking argument as `impl Str` and using `.into() in this functions body.